src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -0,0 +1,1233 @@
+{
+    public partial class SocketAsyncEventArgs : EventArgs, IDisposable
+    {
+        // Struct sizes needed for some custom marshalling.
+        internal static readonly int s_ControlDataSize = Marshal.SizeOf<Interop.Winsock.ControlData>();
+        internal static readonly int s_ControlDataIPv6Size = Marshal.SizeOf<Interop.Winsock.ControlDataIPv6>();
+        internal static readonly int s_WSAMsgSize = Marshal.SizeOf<Interop.Winsock.WSAMsg>();
+
+        // Buffer,Offset,Count property variables.
+        internal WSABuffer m_WSABuffer;
+        internal IntPtr m_PtrSingleBuffer;
+
+        // BufferList property variables.
+        internal WSABuffer[] m_WSABufferArray;
+        private bool _bufferListChanged;
+
+        // Internal buffers for WSARecvMsg
+        private byte[] _WSAMessageBuffer;
+        private GCHandle _WSAMessageBufferGCHandle;
+        internal IntPtr m_PtrWSAMessageBuffer;
+        private byte[] _controlBuffer;
+        private GCHandle _controlBufferGCHandle;
+        internal IntPtr m_PtrControlBuffer;
+        private WSABuffer[] _WSARecvMsgWSABufferArray;
+        private GCHandle _WSARecvMsgWSABufferArrayGCHandle;
+        private IntPtr _ptrWSARecvMsgWSABufferArray;
+
+        // Internal buffer for AcceptEx when Buffer not supplied.
+        internal IntPtr m_PtrAcceptBuffer;
+
+        // Internal SocketAddress buffer
+        private GCHandle _socketAddressGCHandle;
+        private Internals.SocketAddress _pinnedSocketAddress;
+        internal IntPtr m_PtrSocketAddressBuffer;
+        internal IntPtr m_PtrSocketAddressBufferSize;
+
+        // SendPacketsElements property variables.
+        private SendPacketsElement[] _sendPacketsElementsInternal;
+        internal Interop.Winsock.TransmitPacketsElement[] m_SendPacketsDescriptor;
+        internal int m_SendPacketsElementsFileCount;
+        internal int m_SendPacketsElementsBufferCount;
+
+        // Internal variables for SendPackets
+        internal FileStream[] m_SendPacketsFileStreams;
+        internal SafeHandle[] m_SendPacketsFileHandles;
+        internal IntPtr m_PtrSendPacketsDescriptor;
+
+        // Overlapped object related variables.
+        internal SafeNativeOverlapped m_PtrNativeOverlapped;
+        private PreAllocatedOverlapped _preAllocatedOverlapped;
+        private object[] _objectsToPin;
+        private enum PinState
+        {
+            None = 0,
+            NoBuffer,
+            SingleAcceptBuffer,
+            SingleBuffer,
+            MultipleBuffer,
+            SendPackets
+        }
+        private PinState _pinState;
+        private byte[] _pinnedAcceptBuffer;
+        private byte[] _pinnedSingleBuffer;
+        private int _pinnedSingleBufferOffset;
+        private int _pinnedSingleBufferCount;
+
+        internal int? SendPacketsDescriptorCount
+        {
+            get
+            {
+                return m_SendPacketsDescriptor == null ? null : m_sendPacketsDescriptor.Length;
```

Comment:
Fails compilation: needs capital s. I think you meant 

`return m_SendPacketsDescriptor == null ? null : (int?)m_SendPacketsDescriptor.Length;`

Severity	Code	Description	Project	File	Line	Source	Tool
Error	CS0103	The name 'm_sendPacketsDescriptor' does not exist in the current context	System.Net.Sockets	S:\c1\src\System.Net.Sockets\src\System\Net\Sockets\SocketAsyncEventArgs.Windows.cs	87	IntelliSense	Live